helpers/parameter_generator.py

```diff
@@ -171,20 +171,20 @@ def set_one_var_from_string(name, param_type, checks):
     ret = ""
     univar_mapper = {"int": "GetInt", "double": "GetDouble", "bool": "GetBool", "std::string": "GetString"}
     if "vector" not in param_type:
-        ret += "  %s(params, \"%s\", &%s);\n" % (univar_mapper[param_type], name, name)
+        ret += f"  {univar_mapper[param_type]}(params, \"{name}\", &{name});\n"
         if len(checks) > 0:
             check_mapper = {"<": "LT", ">": "GT", "<=": "LE", ">=": "GE"}
             for check in checks:
                 value, sign = parse_check(check)
-                ret += "  CHECK_%s(%s, %s);\n" % (check_mapper[sign], name, value)
+                ret += f"  CHECK_{check_mapper[sign]}({name}, {value});\n"
         ret += "\n"
     else:
-        ret += "  if (GetString(params, \"%s\", &tmp_str)) {\n" % (name)
+        ret += f"  if (GetString(params, \"{name}\", &tmp_str)) {\n"
```

Comment:
Note to self: fix this in the next commit

Reply:
@NovusEdge are you still interested in this PR? Do you need any help?

Reply:
> @NovusEdge are you still interested in this PR?
I am, but I keep running into issues with the linting 😓

Reply:
ok what issues are you facing and what questions can we answer?

Reply:
[this](https://github.com/microsoft/LightGBM/pull/4146/checks?check_run_id=2229633254#step:3:533)

Reply:
What specific question to you have about those checks?

Have you tried reproducing them with the steps in https://github.com/microsoft/LightGBM/pull/4142#pullrequestreview-624538246?